### PR TITLE
[Microsoft.App] optimize java agent related properties

### PR DESCRIPTION
**Pull request by fangjian0423**
1. change `level`type from `string` to `enum`
2. rename `logLevelSettings` to `loggerSettings`

For 2:

Below is a configuration file for logback logging framework:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<configuration>

  <jmxConfigurator />

  <appender name="consoleAppender"
            class="ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36}:%L- !!! %msg%n</pattern>
      <charset class="java.nio.charset.Charset">UTF-8</charset>
    </encoder>
  </appender>

  <logger name="chapters.configuration" level="DEBUG" additivity="false">
    <appender-ref ref="STDOUT" />
  </logger>

  <root level="info">
    <appender-ref ref="consoleAppender" />
  </root>

</configuration>
```

To describe the configurations, the payload could looks like:

```json
{
    "enabled": true,
    "loggerSettings": [
        {
           "logger": "chapters.configuration",
           "level": "debug",
           "additivity":  false,
        }
    ],
    "appenderSettings": [
         // appender list
     ],
     "globalSettings": {
         // jmx
     }
}
```

Currently, we only support level settings for loggers.

### CODE REVIEW

**Comment by najian** [comment]
Detected Breaking change(9) already reviewed in previous PR https://github.com/Azure/azure-rest-api-specs/pull/28405

**Comment by bowen5** [comment]
@ms-zhenhua would you take another look? This PR contains some updates based on https://github.com/Azure/azure-rest-api-specs/pull/28192. BTW, although some breaking change detected, this `2024-02-02-preview` API version for Microsoft.App hasn't been released yet.

**Review by bowen5 — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/examples/ContainerApps_CreateOrUpdate.json [hunk 1]
```diff
@@ -96,7 +96,7 @@
               "enableMetrics": true,
               "javaAgent": {
                 "enabled": true,
-                "logLevelSettings": [
+                "loggerSettings": [
```
Comment (vturecek) [inline specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/examples/ContainerApps_CreateOrUpdate.json:99]: Let's nest this array inside another property to allow space for logging settings that don't apply to specific loggers in the future:

```
"logging": {
  "loggerSettings": [
     "logger": "org.springframework.boot",
     "level": "debug"
  ]
}
```

**Review by bowen5 — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/examples/ContainerApps_CreateOrUpdate.json (hunk 1 above)
Comment (fangjian0423) [reply specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/examples/ContainerApps_CreateOrUpdate.json]: updated

**Pull request merged**